third_party/amd/backend/compiler.py

```diff
@@ -94,6 +94,10 @@ def path_to_rocm_lld():
         lld = Path("/usr/bin/ld.lld")
         if lld.is_file():
             return lld
+        if "TRITON_HIP_LLD_PATH" in os.environ:
```

Comment:
I think typically we do `lld_path = os.getenv("TRITON_HIP_LLD_PATH")`. Also this can be moved to the top to prevail all other options given it's very explicit from the developer.